src/oidcop/configure.py

```diff
@@ -241,8 +227,73 @@ def __init__(
         for key in self.__dict__.keys():
             _val = conf.get(key)
             if not _val:
-                if key in DEFAULT_CONFIG:
-                    _dc = copy.deepcopy(DEFAULT_CONFIG[key])
+                if key in OP_DEFAULT_CONFIG:
+                    _dc = copy.deepcopy(OP_DEFAULT_CONFIG[key])
+                    add_base_path(_dc, base_path, file_attributes)
+                    _val = _dc
+                else:
+                    continue
+            setattr(self, key, _val)
+
+        if self.template_dir is None:
+            self.template_dir = os.path.abspath("templates")
+        else:
+            self.template_dir = os.path.abspath(self.template_dir)
+
+        if not domain:
+            domain = conf.get("domain", "127.0.0.1")
+
+        if not port:
+            port = conf.get("port", 80)
+
+        set_domain_and_port(conf, URIS, domain=domain, port=port)
+
+
+AS_DEFAULT_CONFIG = copy.deepcopy(OP_DEFAULT_CONFIG)
+AS_DEFAULT_CONFIG["claims_interface"] = {
+    "class": "oidcop.session.claims.OAuth2ClaimsInterface", "kwargs": {}}
+
+
+class ASConfiguration(Base):
+    "Authorization server configuration"
+
+    def __init__(
+            self,
+            conf: Dict,
+            base_path: Optional[str] = "",
+            entity_conf: Optional[List[dict]] = None,
+            domain: Optional[str] = "",
+            port: Optional[int] = 0,
+            file_attributes: Optional[List[str]] = None,
+    ):
+
+        conf = copy.deepcopy(conf)
+        Base.__init__(self, conf, base_path, file_attributes)
+
+        self.add_on = None
+        self.authz = None
+        self.authentication = None
+        self.base_url = ""
+        self.capabilities = None
+        self.claims_interface = None
+        self.cookie_handler = None
+        self.endpoint = {}
+        self.httpc_params = {}
+        self.issuer = ""
+        self.keys = None
+        self.session_key = None
+        self.template_dir = None
+        self.token_handler_args = {}
+        self.userinfo = None
```

Comment:
and these are OP's

        self.add_on = None
        self.authz = None
        self.authentication = None
        self.base_url = ""
        self.capabilities = None
        self.cookie_handler = None
        self.endpoint = {}
        self.httpc_params = {}
        self.id_token = None
        self.issuer = ""
        self.keys = None
        self.login_hint2acrs = {}
        self.login_hint_lookup = None
        self.session_key = None
        self.sub_func = {}
        self.template_dir = None
        self.token_handler_args = {}
        self.userinfo = None
        self.password = None
        self.salt = None

They are replicated from OPconfiguration, in a way that if we should add a argument we then need to put in OPConf and at the same time in ASConf. I think that it would be better to have a BaseClass to inherit and take all these attribute from it instead of repeat them

Reply:
Done